src/server/qgsconfigcache.cpp

```diff
@@ -23,27 +22,73 @@
 
 #include <QFile>
 
-QgsConfigCache *QgsConfigCache::instance()
+QgsConfigCache *QgsConfigCache::sInstance = nullptr;
+
+
+void QgsConfigCache::initialize( QgsServerSettings *settings )
 {
-  static QgsConfigCache *sInstance = nullptr;
+  if ( sInstance )
+  {
+    QgsMessageLog::logMessage(
+      QStringLiteral( "Project's cache is already initialized" ),
+      QStringLiteral( "Server" ), Qgis::MessageLevel::Warning );
+    return;
```

Comment:
Crash instead

Reply:
Cf discussion: https://github.com/qgis/QGIS/pull/47050#discussion_r799334397

Reply:
Changed for crashing with Q_ASSERT instead of returning instance with default strategy